### PR TITLE
[Improve] Added Table of Contents and FAQs to README

### PR DESCRIPTION
**Pull request by adarsh-jha-dev**
### Purpose of this pull request

This pull request adds a Table of Contents (TOC) and a Frequently Asked Questions (FAQs) section to the README for better user experience and documentation organization.

### Does this PR introduce _any_ user-facing change?

No, this PR doesn't introduce any user-facing changes. It's a documentation improvement for better readability and accessibility.

### How was this patch tested?

This change was tested by verifying that the documentation renders correctly with the added TOC and FAQs. Manual testing was performed to ensure the TOC links work as expected, providing easy navigation through the document.

### Checklist

* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Thanks @adarsh-jha-dev , Could you fix the CI error?

**Comment by adarsh-jha-dev** [comment]
> Thanks @adarsh-jha-dev , Could you fix the CI error?

yeah sure, on it

**Comment by adarsh-jha-dev** [comment]
> Thanks @adarsh-jha-dev , Could you fix the CI error?

Hey @Hisoka-X ,  I have resolved that CI error, you may proceed further.

**Comment by Hisoka-X** [comment]
cc @EricJoy2048

**Comment by Hisoka-X** [comment]
Could you rebase on dev? We add a new part for SeaTunnel web project.

**Comment by adarsh-jha-dev** [comment]
> Could you rebase on dev? We add a new part for SeaTunnel web project.

it is already having the base branch : dev, do i need to change that? If yes, then to which branch?

**Comment by Hisoka-X** [comment]
rebase on dev so you can resolve conflict. (Or merge from dev).

**Comment by adarsh-jha-dev** [comment]
> rebase on dev so you can resolve conflict. (Or merge from dev).

Done, you may proceed further

File: README.md [hunk 1]
```diff
@@ -6,129 +6,147 @@
 [![Join Slack](https://img.shields.io/badge/slack-%23seatunnel-4f8eba?logo=slack)](https://s.apache.org/seatunnel-slack)
 [![Twitter Follow](https://img.shields.io/twitter/follow/ASFSeaTunnel.svg?label=Follow&logo=twitter)](https://twitter.com/ASFSeaTunnel)
 
----
+SeaTunnel, formerly known as Waterdrop, was rebranded on October 12, 2021.
```
Comment (Hisoka-X) [inline README.md]: Please remove this .

**Comment by Hisoka-X** [comment]
@EricJoy2048 @ic4y @hailin0 @ruanwenjun Are you ok with this change?

File: README.md (hunk 1 above)
Comment (adarsh-jha-dev) [reply README.md]: Done

**Review by EricJoy2048 — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -6,129 +6,145 @@
...
+[Apache 2.0 License](LICENSE)
 
-Please follow this [document](docs/en/contribution/setup.md).
+## Frequently Asked Questions
 
-## Contact Us
+### 1. How do I install SeaTunnel?
 
-- Mail list: **dev@seatunnel.apache.org**. Mail to `dev-subscribe@seatunnel.apache.org`, follow the reply to subscribe to the mail list.
+Follow the [installation guide](https://seatunnel.apache.org/docs/2.1.0/flink/installation/) on our website to get started.
```
Comment (Hisoka-X) [inline README.md]:
```suggestion
Follow the [installation guide](https://seatunnel.apache.org/docs/2.3.3/start-v2/locally/deployment/) on our website to get started.
```

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**